pandas/core/common.py

```diff
@@ -279,7 +279,12 @@ def _isnull_ndarraylike(obj):
     # box
     if isinstance(obj, ABCSeries):
         from pandas import Series
-        result = Series(result, index=obj.index, name=obj.name, copy=False)
+        if isinstance(obj, Series):
+            result = obj._constructor_sliced(
```

Comment:
lots of repeatitive code here. I don't think you need anything except replace Series with ``obj._constructor_sliced`` no?